keras/layers/preprocessing/audio_preprocessing.py

```diff
@@ -0,0 +1,194 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Keras audio preprocessing layers."""
+
+import math
+
+import tensorflow.compat.v2 as tf
+from tensorflow.python.util.tf_export import keras_export
+
+from keras.engine import base_layer
+from keras.engine import base_preprocessing_layer
+from keras.layers.preprocessing import preprocessing_utils as utils
+
+
+@keras_export(
+    "keras.layers.MelSpectrogram",
+    "keras.layers.experimental.preprocessing.MelSpectrogram",
+)
+class MelSpectrogram(base_layer.Layer):
+    """A preprocessing layer to convert raw audio signals to Mel spectrograms.
+
+    This layer takes `float32/float64` single or batched audio signal as inputs
+    and computes the Mel spectrogram using Short-Time Fourier Transform (STFT)
+    and Mel scaling. The input should be a 1D (unbatched) or 2D (batched) tensor
+    representing audio signals. The output will be a 2D or 3D tensor
+    representing Mel spectrograms.
+
+    For an overview and full list of preprocessing layers, see the preprocessing
```

Comment:
Rather than a preprocessing layers reference it would be useful to include info about Mel spectrograms, what they are, why use them, etc. A link to a page would be nice